src/commands/generateBugReport.ts

```diff
@@ -0,0 +1,184 @@
+
+If applicable, add screenshots to help explain your problem.
+
+Expected Behavior
+-----
+
+<!-- What should have happened? -->
+
+Actual Behavior
+-----
+
+<!-- What actually happened? -->
+
+Additional context
+-----
+
+<!--
+Add any other context about the problem here.
+Note whether you use any tools for managing Terraform version/execution (e.g. 'tfenv')
+any credentials helpers, or whether you have any other Terraform extensions installed.
+-->
+`;
+    }
+    const body = `Issue Description
+=====
+
+${problemText}
+
+Environment Information
+=====
+
+Terraform Information
+-----
+
+${this.generateRuntimeMarkdown(this.getRuntimeInfo())}
+
+Visual Studio Code
+-----
+
+| Name | Version |
+| --- | --- |
+| Operating System | ${os.type()} ${os.arch()} ${os.release()} |
+| VSCode | ${vscode.version}|
+
+Visual Studio Code Extensions
+-----
+
+<details><summary>Visual Studio Code Extensions(Click to Expand)</summary>
+
+${this.generateExtesnionMarkdown(extensions)}
+</details>
+
+Extension Logs
+-----
+
+> Find this from the first few lines of the relevant Output pane:
+View -> Output -> 'HashiCorp Terraform'
+
+`;
+    return body;
+  }
+
+  generateExtesnionMarkdown(extensions: VSCodeExtension[]): string {
+    if (!extensions.length) {
+      return 'none';
+    }
+
+    const tableHeader = `|Extension|Author|Version|\n|---|---|---|`;
+    const table = extensions
+      .map((e) => {
+        return `|${e.name}|${e.publisher}|${e.version}|`;
+      })
+      .join('\n');
+
+    const extensionTable = `${tableHeader}\n${table}`;
+    return extensionTable;
+  }
+
+  generateRuntimeMarkdown(info: TerraformInfo): string {
+    const rows = `
+Version:\t${info.version}
+Platform:\t${info.platform}
+Outdated:\t${info.outdated}
+    `;
+    return rows;
+  }
+
+  getExtensions(): VSCodeExtension[] {
+    const extensions = vscode.extensions.all
+      .filter((element) => element.packageJSON.isBuiltin === false)
+      .sort((leftside, rightside): number => {
+        if (leftside.packageJSON.name.toLowerCase() < rightside.packageJSON.name.toLowerCase()) {
+          return -1;
+        }
+        if (leftside.packageJSON.name.toLowerCase() > rightside.packageJSON.name.toLowerCase()) {
+          return 1;
+        }
+        return 0;
+      })
+      .map((ext) => {
+        return {
+          name: ext.packageJSON.name,
+          publisher: ext.packageJSON.publisher,
+          version: ext.packageJSON.version,
+        };
+      });
+    return extensions;
+  }
+
+  getRuntimeInfo(): TerraformInfo {
+    const terraformExe = 'terraform';
+    const cmdargs = ['version', '-json'];
+
+    const spawn = child_process.spawnSync;
+    const child = spawn(terraformExe, cmdargs);
+    const response = child.stdout.toString();
+    const j = JSON.parse(response);
+    return {
+      version: j.terraform_version,
+      platform: j.platform,
+      outdated: j.terraform_outdated,
+    };
```

Comment:
The `json` flag was actually unfortunately only added relatively recently (in 0.13), so passing that to older versions will result in error. See https://github.com/hashicorp/terraform/commit/d8e22f1804fcb99fc70dfbb4f54b9de28a92ebc2

We account for this possibility by falling back to plaintext regex parsing in tfexec https://github.com/hashicorp/terraform-exec/blob/750bc90690f17802eb236adf7336553ea947bcbe/tfexec/version.go#L57-L59
but naturally that's only available for Go, not for TypeScript.

We _could_ source Terraform version via LSP...
 - LS uses tfexec, which has a well tested canonical implementation
 - it most accurately reflects what LS can see (as e.g. `PATH` may differ between VS Code and LS?)

... however it increases the complexity in a bug reporting code path which we probably want to keep as simple as possible. Users may as well be reporting bugs with the LS - in which case we might not able to able obtain the version.

I'm fine with either option - but either way I think we should account for older Terraform versions.

If you decide to keep executing and parsing it from here directly then you can leverage the regular expressions and tests available in tfexec: https://github.com/hashicorp/terraform-exec/blob/750bc90690f17802eb236adf7336553ea947bcbe/tfexec/version.go#L110

Reply:
Great point about the older terraform cli interface, thanks! That's the kind of history I need :)

Adjusted it using your examples to try/catch and do the best it can with parsing strings. I agree, I don't think this feature is worth the effort/complexity of asking the LS, possibly in a case where we are already in an error state.